### PR TITLE
Integration test edge cases

### PR DESCRIPTION
**Pull request by ghost**
This PR addresses #68. 
It also adds a validation rule in the client for the quantity, ensuring that it is always > 0MW. 

More edge cases should still be added, so please suggest some scenarios that you think should still be tested!:)

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 ## New Features
 
-<!-- Here goes the main new features and examples or instructions on how to use them -->
+* Extra validation check to prevent positive quantity
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md]: ?
```suggestion
* Extra validation check to prevent negative quantity
```

File: integration_tests/test_api.py [hunk 2]
```diff
@@ -376,6 +398,120 @@ async def test_stream_gridpool_trades(set_up: dict[str, Any]) -> None:
...
+
+@pytest.mark.asyncio
+async def test_create_order_maximum_price_precision_exceeded(set_up: dict[str, Any]) -> None:
+    """Test creating an order with excessive decimal precision in price."""
+    excessive_precision_price = Price(amount=Decimal("56.123"), currency=Currency.EUR)
+    with pytest.raises(ValueError, match="cannot have more than 2 decimal places"):
+        await create_test_order(set_up, price=excessive_precision_price)
+
+@pytest.mark.asyncio
+async def test_create_order_maximum_quantity_precision_exceeded(set_up: dict[str, Any]) -> None:
```
Comment (nhatcher-frequenz) [inline integration_tests/test_api.py:428]: Should we have those tests also in the service side?

File: integration_tests/test_api.py [hunk 3]
```diff
@@ -376,6 +398,120 @@ async def test_stream_gridpool_trades(set_up: dict[str, Any]) -> None:
...
+            price=set_up["price"],
+            quantity=set_up["quantity"],
+            tag="invalid-delivery-area",
+        )
+    assert excinfo.value.code() == grpc.StatusCode.UNAVAILABLE, "Delivery area not found"
+
+
+@pytest.mark.asyncio
+async def test_create_order_extremely_large_price(set_up: dict[str, Any]) -> None:
+    """Test creating an order with an extremely large price."""
```
Comment (nhatcher-frequenz) [inline integration_tests/test_api.py]: What is the actual max? Can we create two tests:
1. One passing with the Maximum 
2. One failing with Maximum

What we do in our lives echoes eternity

Comment (nhatcher-frequenz) [reply integration_tests/test_api.py]: Is there a maximum quantity?

Should I implement those checks on the service side?

**Review by nhatcher-frequenz — CHANGES_REQUESTED** [review]
Hi @camille-bouvy-frequenz , a couple of minor comments. But LGTM!

File: integration_tests/test_api.py (hunk 2 above)
Comment (ghost) [reply integration_tests/test_api.py:428]: Maybe actually because higher precisions cause handel to fail

File: RELEASE_NOTES.md (hunk 1 above)
Comment (ghost) [reply RELEASE_NOTES.md]: uupsi good call

File: integration_tests/test_api.py (hunk 3 above)
Comment (ghost) [reply integration_tests/test_api.py]: Oh good call I couldn't find minimum & maximum prices on EPEX before but I just found out now that the minimum is -9999 €/MWh  and maximum is 9999 €/MWh. I'll add rules & tests for this. 

For the quantity the minimum is 0.1MWh but I cannot find any maximum (we could implement one ourselves if judged necessary).

**Comment by ghost** [comment]
I now also added validation rules for the quantity and price to be within the bounds from EPEX. The rule about the quantity having to be >= 0.1 MW thus overwrites the rule for positive quantity. I kept both for clarity for now but I could also remove the first rule as it is redundant.

File: integration_tests/test_api.py (hunk 3 above)
Comment (nhatcher-frequenz) [reply integration_tests/test_api.py]: Can you also add a passing test for the maximum price?

Comment (ghost) [reply integration_tests/test_api.py]: Done now ([line 488](https://github.com/frequenz-floss/frequenz-client-electricity-trading-python/pull/70/files#diff-c58504065a8fe72a57213cac8b112171168b3a350c8551881006dc0c7cacadf9R488))!

**Review by nhatcher-frequenz — APPROVED** [review]
LGTM!
